controllers/useraccount/useraccount_controller_test.go

```diff
@@ -493,6 +502,38 @@ func TestReconcile(t *testing.T) {
 			assert.Contains(t, err.Error(), "unable to update status")
 		})
 
+		t.Run("no nstemplateset to create", func(t *testing.T) {
+			userAcc := newUserAccount(username, userID, withoutNSTemplateSet())
+			r, req, cl, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity)
+
+			// when
+			_, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			useraccount.AssertThatUserAccount(t, req.Name, cl).
+				HasConditions(provisioned())
+			AssertThatNSTemplateSet(t, req.Namespace, req.Name, cl).DoesNotExist()
+
+		})
+
+		t.Run("no nstemplateset to update", func(t *testing.T) {
+			userAcc := newUserAccount(username, userID, withoutNSTemplateSet())
+			r, req, cl, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity, preexistingNsTmplSet)
+
+			// when
+			_, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			require.NoError(t, err)
+			useraccount.AssertThatUserAccount(t, req.Name, cl).
+				HasConditions(provisioned())
+			AssertThatNSTemplateSet(t, req.Namespace, req.Name, cl).
+				Exists(). // existed before
+				HasNoOwnerReferences()
+
+		})
```

Comment:
you could change the test to be 100% sure that the NSTemplateset is not touched:
```suggestion
			userAcc := newUserAccount(username, userID, withoutNSTemplateSet(), withFinalizer(toolchainv1alpha1.FinalizerName))
			r, req, cl, _ := prepareReconcile(t, username, userAcc, preexistingUser, preexistingIdentity, preexistingNsTmplSet)
			cl.MockUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
				return fmt.Errorf("schouldn't be called")
			}

			// when
			_, err := r.Reconcile(context.TODO(), req)

			// then
			require.NoError(t, err)
			useraccount.AssertThatUserAccount(t, req.Name, cl).
				HasConditions(provisioned())
			AssertThatNSTemplateSet(t, req.Namespace, req.Name, cl).
				Exists(). // existed before
				HasNoOwnerReferences().
				HasNamespaceTemplateRefs(devTemplateRef,codeTemplateRef).
				HasClusterResourcesTemplateRef(clusterResourcesTemplateRef)
		})
```